python/lsst/daf/butler/remote_butler/_remote_butler.py

```diff
@@ -0,0 +1,317 @@
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This software is dual licensed under the GNU General Public License and also
+# under a 3-clause BSD license. Recipients may choose which of these licenses
+# to use; please see the files gpl-3.0.txt and/or bsd_license.txt,
+# respectively.  If you choose the GPL option then the following text applies
+# (but note that there is still no warranty even if you opt for BSD instead):
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from collections.abc import Collection, Iterable, Sequence
+from contextlib import AbstractContextManager
+from typing import Any, TextIO
+
+import httpx
+from lsst.daf.butler import __version__
+from lsst.resources import ResourcePath, ResourcePathExpression
+from lsst.utils.introspection import get_full_type_name
+
+from .._butler import Butler
+from .._butler_config import ButlerConfig
+from .._config import Config
+from .._dataset_existence import DatasetExistence
+from .._dataset_ref import DatasetIdGenEnum, DatasetRef
+from .._dataset_type import DatasetType
+from .._deferredDatasetHandle import DeferredDatasetHandle
+from .._file_dataset import FileDataset
+from .._limited_butler import LimitedButler
+from .._storage_class import StorageClass
+from ..datastore import DatasetRefURIs
+from ..dimensions import DataId, DimensionConfig, DimensionUniverse
+from ..registry import Registry, RegistryDefaults
+from ..transfers import RepoExportContext
+from ._config import RemoteButlerConfigModel
+
+
+class RemoteButler(Butler):
+    def __init__(
+        self,
+        # These parameters are inherited from the Butler() constructor
+        config: Config | ResourcePathExpression | None = None,
+        *,
+        collections: Any = None,
+        run: str | None = None,
+        searchPaths: Sequence[ResourcePathExpression] | None = None,
+        writeable: bool | None = None,
+        inferDefaults: bool = True,
+        # Parameters unique to RemoteButler
+        http_client: httpx.Client | None = None,
+        **kwargs: Any,
+    ):
+        butler_config = ButlerConfig(config, searchPaths, without_datastore=True)
+        self._config = RemoteButlerConfigModel.model_validate(butler_config)
```

Comment:
Does this validate because `ButlerConfig` looks like a generic mapping and it happens to have a `remote_butler.url` in the hierarchy? In general `ButlerConfig` is not a pydantic model even though we have talked about it being one. What happens to all the other parts of `ButlerConfig`?

Reply:
Yes, that's correct.  Currently the other parts of ButlerConfig are being discarded since we're not using them.  I figured as we added usage of the other parts of ButlerConfig, we could add the appropriate validation for them. 

It's still not clear to me which parts of the existing DirectButler ButlerConfig are generated internally in the client code, which are potentially configured locally in a client-side config file, and which are only server-side concepts that the client never sees in a configuration file.

I do wonder if some portion of the configuration would be returned from a versioned "init" endpoint (which maybe gives you back some config chunks, server capabilities, dimensions, collection names, etc all in one shot.)  If there is only one "config file" endpoint that configuration format becomes an unversionable permanent part of the API